src/lib/builder/Web3FunctionBuilder.spec.ts

```diff
@@ -0,0 +1,137 @@
+import path from "node:path";
+import { Web3FunctionBuilder } from "./Web3FunctionBuilder";
+
+describe("Web3FunctionBuilder.build", () => {
+  const TEST_FOLDER_BASE = path.join(
+    process.cwd(),
+    "src/lib/builder/__test__/"
+  );
+
+  const buildTestPath = (folder: string): string => {
+    return path.join(TEST_FOLDER_BASE, folder);
+  };
+
+  const buildSchemaPath = (folder: string): string => {
+    return path.join(buildTestPath(folder), "index.ts");
+  };
+
+  test("should fail when input path does not exist", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("not-existing")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes("Missing Web3Function schema")
+      ).toBeTruthy();
```

Comment:
why not use this more compact matcher instead:
```suggestion
    expect(res.error.message).toMatch("Missing Web3Function schema");
```

Reply:
I thought is was checking for exact match, 5a3b57652d8bc12a6d4f849b1f72c2fcb8ac7eee